spring-cloud-generator/showcase/showcase-spring-starter/src/main/java/com/google/showcase/v1beta1/spring/EchoSpringAutoConfiguration.java

```diff
@@ -130,6 +130,7 @@ public EchoSettings echoSettings(
     clientSettingsBuilder
         .setCredentialsProvider(this.credentialsProvider)
         .setTransportChannelProvider(defaultTransportChannelProvider)
+        .setEndpoint(EchoSettings.getDefaultEndpoint())
```

Comment:
Not sure if this is what we want when creating client settings. We can follow up with proper changes in gax. 
cc: @lqiu96

Reply:
Do you have the error message if you don't add this line to set the endpoint? Ideally, we shouldn't have to set the endpoint ourselves but for this PR, I think it's fine we can't resolve the issue here.

Reply:
> Do you have the error message if you don't add this line to set the endpoint?

The error message:
```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'echoClient' defined in com.google.showcase.v1beta1.spring.EchoSpringAutoConfiguration: Failed to instantiate [com.google.showcase.v1beta1.EchoClient]: Factory method 'echoClient' threw exception with message: Invalid host or port: .googleapis.com 443
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:648)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:636)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1335)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1165)
```

It fails to resolve the endpoint if we don't explicitly set it: `Invalid host or port: .googleapis.com 443`

> Or if this isn't because of an error, is this because we're using an old version of the generator. Or perhaps this is just something spring generator does?

I don't think this is very different from a normal use case of instantiating a client with a few settings.

Here we create the settings:
https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/0f94e07471093fd5a2165907a0a4bd3c006ffd0d/spring-cloud-generator/showcase/showcase-spring-starter/src/main/java/com/google/showcase/v1beta1/spring/EchoSpringAutoConfiguration.java#L115-L121

Here we create the client with these settings:
https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/0f94e07471093fd5a2165907a0a4bd3c006ffd0d/spring-cloud-generator/showcase/showcase-spring-starter/src/main/java/com/google/showcase/v1beta1/spring/EchoSpringAutoConfiguration.java#L339-L343

Reply:
> It fails to resolve the endpoint if we don't explicitly set it: Invalid host or port: .googleapis.com 443

Ah, I think this is an edge case due to the the showcase protos. The showcase protos have a `default_host` value that won't match a "normal" google endpoint. For example, the echo proto's `service_host` value: https://github.com/googleapis/gapic-showcase/blob/4e7c30a5d5df8aac30e8b8f1e6f71809fa9d20f6/schema/google/showcase/v1beta1/echo.proto#L45
 where as a normal endpoint would be something like `cloudassets.googleapis.com`.

The error above is complaining that we can't populate the `serviceName` part of the endpoint and that's because the showcase protos don't have one. That's why you'd have to explicitly call `setEndpoint()` yourself.

This looks like it's changing the code inside the generator, which I believe would impact all of spring code gen. The endpoint normally *should* be constructed internally for non-showcase services and I believe having the generator explicitly call `setEndpoint()` may be imcompatible with enabling certain features.

In the showcase ITs, we have this same problem. But we solve it by explicitly calling `setEndpoint()` in the handwritten layers: https://github.com/googleapis/sdk-platform-java/blob/5799827a864be627bac03969cc178efc9d6170aa/showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/util/TestClientInitializer.java#L67

I'm not too familiar with the showcase autogen ITs. Are there any handwritten parts that we can do something similar or are these the only tests that are generated + run?

Reply:
> Are there any handwritten parts that we can do something similar or are these the only tests that are generated + run?

This is the [handwritten test](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/5b9fb60ca29e43d5ccac954dfd8fefca80715468/spring-cloud-generator/showcase/showcase-spring-starter/src/test/java/com/google/showcase/v1beta1/spring/EchoAutoConfigurationTests.java).Compare to GAPIC showcase tests, this test though does not create a client, instead, it fetches the client from existing Spring beans. So it's hard to override the endpoint in the tests since the whole point of SpringCodeGen is to create a client for customers.

I think this change might be unavoidable for now, and should not impact any existing features, because SpringCodeGen does not expose a property for customers to set endpoint or universe domain either. We can revisit this decision once we get to implement universe domain in spring-cloud-gcp. WDYT @lqiu96 ?

Reply:
I see. I'm not too familiar with spring so I don't know of a way to do this, but we are going to have to remove the explicit setting for the default endpoint at some point (i.e. whenever we enable universe domain). This will require a workaround for the showcase clients in the future or more configurations generated by the generator.

I think for now we can proceed with explicitly setting the default endpoint.